tests/models/chameleon/test_processor_chameleon.py

```diff
@@ -0,0 +1,33 @@
+import tempfile
+import unittest
+
+from transformers import ChameleonProcessor
+from transformers.models.auto.processing_auto import processor_class_from_name
+
+from ...test_processing_common import ProcessorTesterMixin
+
+
+class ChameleonProcessorTest(ProcessorTesterMixin, unittest.TestCase):
+    from_pretrained_id = "leloy/Anole-7b-v0.1-hf"
+    processor_class = ChameleonProcessor
+
+    def get_component(self, attribute, **kwargs):
+        assert attribute in self.processor_class.attributes
+        component_class_name = getattr(self.processor_class, f"{attribute}_class")
+        if isinstance(component_class_name, tuple):
+            if "_fast" in component_class_name[0]:
+                component_class_name = component_class_name[0]
+            else:
+                component_class_name = component_class_name[1]
```

Comment:
Same question as in the other PR, why we need to overwrite and look for fastTokenizer? Or is it FastImageProcessor

Reply:
some of the common tests error-out with the base tokenizer

I've yet to investigate why, but it's likely unrelated to this PR

Reply:
Yes, nice to see what exactly is causing the error, in case there is a bug in tokenizers

Reply:
ah, I remember now

The slow, LlamaTokenizer expects the vocab file to be present but it's neither in the official repo nor does it get saved to the temp dir when we do `processor.save_pretrained(self.tmpdirname)` in `setUp`

imma add this as a comment

Reply:
Ah right, chameleon never had a slow tokenizer. Oke, in that case we can and prob should remove an option for slow tokenizer in chameleon here so that the tuple is `(None, FastTokenizer)`

https://github.com/huggingface/transformers/blob/85345bb439652d3f03bb4e123cef7a440f2ba95b/src/transformers/models/auto/tokenization_auto.py#L111-L115

And then add a check for `Noneness` in general `get_component`

Reply:
just removed Chameleon's slow tokenizer & removed the custom `get_component` in `ChameleonProcessorTest` (we don't need the extra check cuz there's only one tokenizer left